appinventor/blocklyeditor/src/blocks/components.js

```diff
@@ -170,6 +170,13 @@ Blockly.Blocks.component_event = {
     if (this.instanceName == oldname) {
       this.instanceName = newname;
       this.componentDropDown.setValue(this.instanceName);
+      if (this.isCollapsed())
+      {
+        var COLLAPSED_INPUT_NAME = '_TEMP_COLLAPSED_INPUT';     
```

Comment:
Is there anything else I should change?

There's an appspot instance with this feature: http://ai2-experiments.appspot.com/

Thanks!